### PR TITLE
Add flow metric for IP address and port number together

### PR DESCRIPTION
**Pull request by leoparente**

### CODE REVIEW

**Review by rboucher-me — APPROVED** [review]
Looks good to me.

File: src/handlers/flow/FlowStreamHandler.cpp [hunk 1]
```diff
@@ -582,8 +614,18 @@ void FlowMetricsBucket::process_flow(bool deep, const FlowPacket &payload)
 
         if (!flow.is_ipv6 && flow.ipv4_out.isValid()) {
             group_enabled(group::FlowMetrics::Cardinality) ? _dstIPCard.update(flow.ipv4_out.toInt()) : void();
-            group_enabled(group::FlowMetrics::TopByBytes) ? _topByBytes.topDstIP.update(flow.ipv4_out.toString(), flow.payload_size) : void();
-            group_enabled(group::FlowMetrics::TopByPackets) ? _topByPackets.topDstIP.update(flow.ipv4_out.toString(), flow.packets) : void();
+            if (group_enabled(group::FlowMetrics::TopByBytes)) {
+                _topByBytes.topDstIP.update(flow.ipv4_out.toString(), flow.payload_size);
+                if (flow.src_port > 0) {
```
Comment (weyrick) [inline src/handlers/flow/FlowStreamHandler.cpp]: dst_port

File: src/handlers/flow/FlowStreamHandler.cpp [hunk 2]
```diff
@@ -596,8 +638,18 @@ void FlowMetricsBucket::process_flow(bool deep, const FlowPacket &payload)
             }
         } else if (flow.is_ipv6 && flow.ipv6_out.isValid()) {
             group_enabled(group::FlowMetrics::Cardinality) ? _dstIPCard.update(reinterpret_cast<const void *>(flow.ipv6_out.toBytes()), 16) : void();
-            group_enabled(group::FlowMetrics::TopByBytes) ? _topByBytes.topDstIP.update(flow.ipv6_out.toString(), flow.payload_size) : void();
-            group_enabled(group::FlowMetrics::TopByPackets) ? _topByPackets.topDstIP.update(flow.ipv6_out.toString(), flow.packets) : void();
+            if (group_enabled(group::FlowMetrics::TopByBytes)) {
+                _topByBytes.topDstIP.update(flow.ipv6_out.toString(), flow.payload_size);
+                if (flow.src_port > 0) {
```
Comment (weyrick) [inline src/handlers/flow/FlowStreamHandler.cpp]: dst_port

**Review by weyrick — CHANGES_REQUESTED** [review]

File: src/handlers/flow/FlowStreamHandler.cpp (hunk 1 above)
Comment (leoparente) [reply src/handlers/flow/FlowStreamHandler.cpp]: done

File: src/handlers/flow/FlowStreamHandler.cpp (hunk 2 above)
Comment (leoparente) [reply src/handlers/flow/FlowStreamHandler.cpp]: done

File: src/handlers/flow/FlowStreamHandler.h [hunk 3]
```diff
@@ -79,13 +79,17 @@ class FlowMetricsBucket final : public visor::AbstractMetricsBucket
...
+        TopN<std::string> topSrcIPandPort;
+        TopN<std::string> topDstIPandPort;
         TopN<uint32_t> topInIfIndex;
         TopN<uint32_t> topOutIfIndex;
         topns(std::string metric)
             : topSrcIP("flow", "ip", {"top_src_ips_" + metric}, "Top source IP addresses by " + metric)
             , topDstIP("flow", "ip", {"top_dst_ips_" + metric}, "Top destination IP addresses by " + metric)
             , topSrcPort("flow", "port", {"top_src_ports_" + metric}, "Top source ports by " + metric)
             , topDstPort("flow", "port", {"top_dst_ports_" + metric}, "Top destination ports by " + metric)
+            , topSrcIPandPort("flow", "ip_port", {"top_src_ips_and_port_" + metric}, "Top source IP addresses and port by " + metric)
```
Comment (leoparente) [inline src/handlers/flow/FlowStreamHandler.h:91]: @weyrick  @rboucher-ns1  do you agree with this tag name "ip_port" instead of only "ip"

Comment (weyrick) [reply src/handlers/flow/FlowStreamHandler.h:91]: yes, it should definitely not just be "ip" or just "port" so i guess this is best

**Review by weyrick — APPROVED** [review]

**Pull request merged**